system-addon/lib/TopSitesFeed.jsm

```diff
@@ -366,10 +366,14 @@ this.TopSitesFeed = class TopSitesFeed {
       // All these actions mean we need new top sites
       case at.MIGRATION_COMPLETED:
       case at.PLACES_HISTORY_CLEARED:
-      case at.PLACES_LINKS_DELETED:
         this.frecentCache.expire();
         this.refresh({broadcast: true});
         break;
+      case at.PLACES_LINKS_CHANGED:
+        this.frecentCache.expire();
+        this.pinnedCache.expire();
```

Comment:
I don't think pinned cache needs to expire from thees history/bookmark places changes… ? I.e., are there any PLACES_LINKS_CHANGED events that would cause the pinned cache to be wrong?